### ISSUE TITLE
no published sdist for 2.4.2

### ISSUE THREAD

**Issue by dhellmann** [mjpost/sacrebleu#269]
I'm trying to prepare a build of sacrebleu from source but there is no sdist on pypi for the 2.4.2 release. Is that an oversight or are you no longer publishing sdists?

**Comment by martinpopel** [comment]
I confirm https://pypi.org/project/sacrebleu/2.4.2/#files is missing `sacrebleu-2.4.2.tar.gz` (it includes only `sacrebleu-2.4.2-py3-none-any.whl`). I guess @mjpost will confirm it is an oversight.
Meanwhile, you can obtain [https://github.com/mjpost/sacrebleu/archive/refs/tags/v2.4.2.tar.gz](v2.4.2.tar.gz) from [GitHub](https://github.com/mjpost/sacrebleu/releases).
BTW: sacrebleu is a pure-Python package (no compilation of C/C++ is needed), so even the `whl` file contains (just) all the source codes.

**Comment by dhellmann** [comment]
Thanks, @martinpopel. [The tool we're working with](https://github.com/python-wheel-build/fromager) automates builds from sdists, so I thought it was worth checking and I appreciate the quick response. We can do some work to configure it to build from that source archive if we have to, but if you consider it an oversight I may just wait a day or so to see if you have time to upload the sdist.

I'd be happy to help you set up a GitHub action for publishing releases from tags, too. I have it configured in several repos, and find it really simplifies things. If you're interested, I could send you a PR and help you debug any issues setting up the authentication. Let me know.

**Comment by mjpost** [comment]
Hi @dhellmann, this would be great to have. We'd be very grateful you'd contribute a PR and tell us how to make use of the feature. It's one of those things I can never seem to make time for.

**Comment by dhellmann** [comment]
Great, I'll be happy to help. Today is pretty full, but I should have time later this week to prepare a PR for you based on what I do in other repos like this: https://github.com/python-wheel-build/fromager/blob/main/.github/workflows/python-publish.yaml

There's a link in the top of that file that points to the GitHub.com docs for publishing python packages.

The main thing that I stumbled over when setting up the trusted publishing hook was failing to set the ["environment" value](https://github.com/python-wheel-build/fromager/blob/main/.github/workflows/python-publish.yaml#L15).

**Comment by ozancaglayan** [comment]
cc @martinpopel for decision

**Comment by mjpost** [comment]
I think so, with @dhellmann's PR from some time ago. Thanks for following up, @ozancaglayan.

**Comment by martinpopel** [comment]
Yes, I confirm https://pypi.org/project/sacrebleu/2.6.0/#files contains also the sources (
sacrebleu-2.6.0.tar.gz) in addition to the wheel (sacrebleu-2.6.0-py3-none-any.whl), so this issue is solved.
Thanks for watching this issue (@ozancaglayan), closing it (@mjpost) and most importantly to @dhellmann for implementing the GitHub action for publishing releases to PyPI (I've reused it in another Python project I maintain).

**Comment by dhellmann** [comment]
Thanks for taking the PR, and I'm glad the technique has proven useful elsewhere. :-)
